scripts/weekly-dep-report.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+
+# this script is making an assumption that the repository contains at least two commits
+# and commits span across at least two weeks
+
+# expects a parameter that is the name of the default branch, i.e., "main"
+starter_point=${1:-"HEAD"}
+git checkout --quiet "${starter_point}"
+
+last_commit=$(git rev-list HEAD -1)
+first_commit_within_last_seven_days=$(git rev-list --reverse --since=7.days.ago HEAD | head -n 1)
+
+format_change_in_dependency_output () {
+    # if negative, do nothing; if positive, add + prefix; if zero, return "="
+    if [[ $1 > 0 ]]; then
+        echo "+$1"
+    elif [[ $1 == 0 ]]; then
+        echo "="
+    else
+        echo "$1"
+    fi
+}
+
+get_total_deps () {
+    echo $(cargo +stable guppy resolve-cargo --resolver-version v2 --target-platform x86_64-unknown-linux-gnu --host-platform x86_64-unknown-linux-gnu --build-kind target --kind thirdparty --omit-edges-into diem-workspace-hack | wc -l | tr -d ' ')
```

Comment:
After testing in CI, lets remove the `+stable` from all of these commands and just use the cargo version from the pinned toolchain